driver/ppm_events_public.h

```diff
@@ -1606,22 +1606,36 @@ enum extra_event_prog_code
 	PPM_SC_X(EPOLL_WAIT_OLD, 379) \
 	PPM_SC_X(RSEQ, 380) \
 	PPM_SC_X(CREATE_MODULE, 381) \
-	PPM_SC_X(PIDFD_GETFD, 382) \
-	PPM_SC_X(SCHED_GETATTR, 383) \
-	PPM_SC_X(FACCESSAT2, 384) \
-	PPM_SC_X(_SYSCTL, 385) \
-	PPM_SC_X(QUERY_MODULE, 386) \
-	PPM_SC_X(GET_MEMPOLICY, 387) \
-	PPM_SC_X(SYNC_FILE_RANGE, 388) \
-	PPM_SC_X(PROCESS_MADVISE, 389) \
-	PPM_SC_X(MEMBARRIER, 390) \
-	PPM_SC_X(IOPL, 391) \
-	PPM_SC_X(CLOSE_RANGE, 392) \
-	PPM_SC_X(FANOTIFY_MARK, 393)
+	PPM_SC_X(SCHED_GETATTR, 382) \
+	PPM_SC_X(FACCESSAT2, 383) \
+	PPM_SC_X(_SYSCTL, 384) \
+	PPM_SC_X(QUERY_MODULE, 385) \
+	PPM_SC_X(GET_MEMPOLICY, 386) \
+	PPM_SC_X(SYNC_FILE_RANGE, 387) \
+	PPM_SC_X(PROCESS_MADVISE, 388) \
+	PPM_SC_X(MEMBARRIER, 389) \
+	PPM_SC_X(IOPL, 390) \
+	PPM_SC_X(CLOSE_RANGE, 391) \
+	PPM_SC_X(FANOTIFY_MARK, 392)
+
+// Tracepoint based events start from 1 << 10 (1024)
+#define PPM_SC_TP_SHIFT 10
+#define PPM_SC_TP_START (1<<PPM_SC_TP_SHIFT)
+#define PPM_SC_TP_X PPM_SC_X // this is used to help syscalls-bumper (https://github.com/falcosecurity/syscalls-bumper) automation
+#define PPM_SC_TP_FIELDS \
+	PPM_SC_TP_X(SCHED_PROCESS_EXIT, PPM_SC_TP_START+0) \
+	PPM_SC_TP_X(SCHED_SWITCH, PPM_SC_TP_START+1) \
+	PPM_SC_TP_X(PAGE_FAULT_USER, PPM_SC_TP_START+2) \
+	PPM_SC_TP_X(PAGE_FAULT_KERNEL, PPM_SC_TP_START+3) \
+	PPM_SC_TP_X(SIGNAL_DELIVER, PPM_SC_TP_START+4) \
```

Comment:
Here we must list just the `PPM_SC_` coming from a tracepoint (eg: "sched_switch" or any LSM/kprobe we might add in the future).

This is just needed to ensure an easy life  to syscalls-bumper. Moreover, we can easily iterate on syscalls only.
I think that for syscalls-bumper alone, it should not be really an issue if this was a list of ppm sc coming from both syscalls and tracepoints mixed.
But then, we would lose the ability to iterate on syscalls only events.

Reply:
Perhaps we might want to investigate whether we really need to iterate over just syscalls/just tracepoints.

Reply:
@gnosek from my understanding I think this is exactly what this enum is doing. Here we're enumerating the `procexit` event, whereas the knowledge about which tracepoint needs to be attached is kept internal in the engines.

Perhaps, maybe it can help to name these constants in a less ambiguous way?

Reply:
Agree with Jason; the user must be able to choose the `SCHED_PROCESS_EXIT` event, therefore we expose it through our ppm sc enum